storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/sql/KafkaSqlStore.java

```diff
@@ -0,0 +1,158 @@
+ * only needed by the KSQL storage.
+ * 
+ * @author eric.wittmann@gmail.com
+ */
+@ApplicationScoped
+@Named("KafkaSqlStore")
+public class KafkaSqlStore extends AbstractSqlRegistryStorage {
+
+    private static final Logger log = LoggerFactory.getLogger(KafkaSqlStore.class);
+
+    @PostConstruct
+    void onConstruct() {
+        log.info("Using internal SQL storage.");
+    }
+
+    public <R, X extends Exception> R withHandle(HandleCallback<R, X> callback) {
+        try {
+            return this.jdbi.withHandle(callback);
+        } catch (Exception e) {
+            throw new RegistryStorageException(e);
+        }
+    }
+
+    public boolean isArtifactExists(String artifactId) throws RegistryStorageException {
+        return withHandle( handle -> {
+            String sql = sqlStatements().selectArtifactCountById();
+            return handle.createQuery(sql)
+                    .bind(0, artifactId)
+                    .mapTo(Integer.class)
+                    .one() > 0;
+        });
+    }
+
+    public boolean isContentExists(String contentHash) throws RegistryStorageException {
+        return withHandle( handle -> {
+            String sql = sqlStatements().selectContentCountByHash();
+            return handle.createQuery(sql)
+                    .bind(0, contentHash)
+                    .mapTo(Integer.class)
+                    .one() > 0;
+        });
+    }
+
+    public boolean isArtifactRuleExists(String artifactId, RuleType rule) throws RegistryStorageException {
+        return withHandle( handle -> {
+            String sql = sqlStatements().selectArtifactRuleCountByType();
+            return handle.createQuery(sql)
+                    .bind(0, artifactId)
+                    .bind(1, rule.name())
+                    .mapTo(Integer.class)
+                    .one() > 0;
+        });
+    }
+
+    public boolean isGlobalRuleExists(RuleType rule) throws RegistryStorageException {
+        return withHandle( handle -> {
+            String sql = sqlStatements().selectGlobalRuleCountByType();
+            return handle.createQuery(sql)
+                    .bind(0, rule.name())
+                    .mapTo(Integer.class)
+                    .one() > 0;
+        });
+    }
+    
+    public void storeContent(String contentHash, ArtifactType artifactType, ContentHandle content) throws RegistryStorageException {
+        withHandle( handle -> {
+            super.createOrUpdateContent(handle, artifactType, content);
+            return null;
+        });
+    }
+    
+    private ContentHandle getContent(long contentId) {
+        return withHandle( handle -> {
+            String sql = sqlStatements().selectContentById();
+            return handle.createQuery(sql)
+                    .bind(0, contentId)
+                    .map(ContentMapper.instance)
+                    .one();
+        });
+    }
+
+    public CompletionStage<ArtifactMetaDataDto> createArtifactWithMetadata(String artifactId,
```

Comment:
The parent method that this method is calling is annotated. I don't think this is needed.

Reply:
Sorry, you were right, here we're also calling the protected method without annotation.

Reply:
*technically* this does not need to be transactional right now, because writes are single-threaded.  But still, it is more logically correct and consistent to make it transactional, so I will do that. :)